ctrd/utils.go

```diff
@@ -166,7 +165,8 @@ func (c *Client) getResolver(ctx context.Context, authConfig *types.AuthConfig,
 	}
 
 	if availableRef == "" {
-		return nil, "", fmt.Errorf("there is no available image reference after trying %+q", refs)
+		logrus.Errorf("there is no available image reference after trying %+q", refs)
+		return nil, "", errtypes.ErrNotfound
```

Comment:
prefer `errors.Wrapf(errtypes.ErrNotfound, "there is no available image reference after trying %+q", refs)` here
util function just directly return error, log or not should be decided by upper.

Reply:
I change to `logrus.warnf`, add an another log in upperand. But I think `errtypes.ErrNotfound` is enough here, user didn't care about how you find a available ref, just tell them not found.

Reply:
make sense